### PR TITLE
`Hds::Icon` Groundwork

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

This PR is part of https://github.com/hashicorp/design-system/pull/2207

If merged, this PR will:

- Install `@hashicorp/flight-icons` as a direct dependency of `components`
- Port the load-sprite instance initialization functionality over from `ember-flight-icons` to `components`
- Ensure that the load-sprite initializer only runs once. (In existing and new instance initializers)

### :link: External links

Feature branch PR: [2207](https://github.com/hashicorp/design-system/pull/2207)
Jira ticket: [HDS-3515](https://hashicorp.atlassian.net/browse/HDS-3515)

***

[HDS-3515]: https://hashicorp.atlassian.net/browse/HDS-3515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: yarn.lock
Comment (didoo) [inline yarn.lock:4120]: current latest version is `3.4.0` (hence the double entry in `yarn.lock` I think)
https://www.npmjs.com/package/@hashicorp/flight-icons?activeTab=versions

**Comment by Dhaulagiri** [comment]
It doesn't look like the logic for avoiding double insertion of the sprite is working as I see it twice in the website 👀 

Screenshot 2024-07-03 at 09 24 39

**Comment by zamoore** [comment]
@Dhaulagiri I'm unable to reproduce the double loading locally. Maybe something environment related?

**Comment by didoo** [comment]
@zamoore see the deployed preview website: https://hds-website-git-hdsicon-importicons-hashicorp.vercel.app/

screenshot_3919

**Comment by zamoore** [comment]
@didoo @Dhaulagiri - I had to make the same change to addon-main that I made to the load-sprite initializer for cases where that isn't used. Checked the preview now and it does appear to only load once.

**Review by didoo — APPROVED** [review]
LGTM

(let's wait for @Dhaulagiri or @alex-ju to approve too, for this one)

File: packages/components/addon-main.cjs
Comment (alex-ju) [inline packages/components/addon-main.cjs:14]: note: I assume the double underscore notation is used to denote the 'internal' nature of this property. across the codebase we have all sorts of notations for this, from `__PROPERTY__` to `_property` and `__property` that would ideally get aligned in the future

**Review by alex-ju — APPROVED** [review]
Code looks good! I'm happy for us to get it in, assuming we tested this in a product as well.

**Comment by didoo** [comment]
@zamoore do you think we should do a quick check in Cloud UI using this specific branch, to make sure we're not missing anything? unless you're 100% confident is not necessary to retest what happens if we deploy only this groundwork first

I'll leave it to you to make the call :)

**Comment by zamoore** [comment]
@didoo I'd rather be more cautious than less. I'm going to run a quick test this morning to verify.

File: packages/components/addon-main.cjs
Comment (zamoore) [reply packages/components/addon-main.cjs:14]: What do you think the best way to document this would be?

**Pull request merged**
